tests/meltano/core/test_project_init_service.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import platform
+import stat
+from typing import TYPE_CHECKING
+
+import pytest
+
+from meltano.core.project_init_service import (
+    ProjectInitService,
+    ProjectInitServiceError,
+)
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+
+def test_project_init_directory_exists(tmp_path: Path, pushd):
+    project_name = "test_project"
+
+    projects_dir = tmp_path.joinpath("exists")
+    projects_dir.joinpath(project_name).mkdir(parents=True)
+    pushd(projects_dir)
+    with pytest.raises(
+        ProjectInitServiceError,
+        match="Directory test_project already exists",
+    ):
+        ProjectInitService(project_name).init(activate=False, add_discovery=False)
+
+
+def test_project_init_no_write_permission(tmp_path: Path, pushd):
+    project_name = "test_project"
```

Comment:
Ok this is a bit painful 🙃

```suggestion
    if platform.system() == "Windows":
        pytest.xfail(
            "Windows can still create new directories inside a read-only directory."
        )
    project_name = "test_project"
```

cc @WillDaSilva in case you know if there's a way to trigger a permission error on Windows.